components/brave_wallet/browser/solana_transaction.h

```diff
@@ -51,21 +51,20 @@ class SolanaTransaction {
     std::optional<bool> skip_preflight;
   };
 
-  explicit SolanaTransaction(SolanaMessage&& message);
+  explicit SolanaTransaction(SolanaMessage message);
 
   SolanaTransaction(
       mojom::SolanaMessageVersion version,
       const std::string& recent_blockhash,
       uint64_t last_valid_block_height,
       const std::string& fee_payer,
       const SolanaMessageHeader& message_header,
-      std::vector<SolanaAddress>&& static_account_keys,
```

Comment:
What is the reason of changing this? It forces callers to move checked by compiler and has better performance.
Even if the caller uses move, it still does a copy construct. Compare this simple example and the instructions it generates
https://godbolt.org/z/vs1G69947 vs https://godbolt.org/z/K5Mz1hYqc

Reply:
This guide https://google.github.io/styleguide/cppguide.html#Rvalue_references discourages use of r-value references for arguments.
Godbolt comparisons should use clang with args at least including `-O2 -fno-exceptions` but this immediately over optimizes these examples without obvious difference.